docs/next/guides/building-and-testing/plugins.md

```diff
@@ -1,242 +1,175 @@
 canonicalUrl: /plugins
 tags:
   - custom plugin
   - skeleton
-  - internal
-  - external
   - extend
 ---
 
 # Plugins
 
 ## Overview
 
-Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of features available in tis library are provided by plugins.
+Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of features available in this library are provided by plugins.
 
-### Plugin template
+This guide will lead you, step-by-step, through the process of creating a custom extension.
 
-There are two types of plugins: internal and external. While both extend Handsontable's functionality, the former is incorporated into the Handsontable build and the latter needs to be included from a separate file.
+### 1. Prerequisites
 
-Regardless of which plugin you are going to build, using a template will save you lots of time.
+You need to import the following things:
+- `BasePlugin` - contains the basic interface to work within Handsontable's lifecycle,
+- `registerPlugin` - utility to register the plugin in Handsontable plugins registry.
 
-#### Internal plugin
+Both provide the basic interface necessary to create a reusable plugin.
+
+```js
+import { BasePlugin, registerPlugin } from 'handsontable/plugins';
+```
+
+### 2. Extend BasePlugin
+Extending our `BasePlugin` is a recommended way to start creating your next extension.
+In this built-in interface, we care about many factors such as backward compatibility, memory leaks preventions and proper binding plugin's instance with Handsontable.
 
-The source code of this template is [available on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/internal). Start off by cloning the entire project or copying the code to your application.
 
 ```js
-// You need to import the BasePlugin class in order to inherit from it.
-import BasePlugin from './../../_base';
-import {registerPlugin} from './../../../plugins';
-
-/**
- * @plugin InternalPluginSkeleton
- * Note: keep in mind, that Handsontable instance creates one instance of the plugin class.
- *
- * @description
- * Blank plugin template. It needs to inherit from the BasePlugin class.
- */
-class InternalPluginSkeleton extends BasePlugin {
-
-  // The argument passed to the constructor is the currently processed Handsontable instance object.
+export class CustomPlugin extends BasePlugin {
+  /**
+   * Define an unique plugin's key.
+   * Handsontable will use that string to register the plugin under that alias.
+   * Moreover, you might also use the same parameter to recognize the plugin's
+   * options passed through the setting object at Handsontable initialization.
+   *
+   * @returns {string}
+   */
+  static get PLUGIN_KEY() {
+    return 'customPlugin';
+  }
+
+  /**
+   * Extend the default constructor and define internal properties for your plugin.
+   *
+   * @param {Handsontable} hotInstance
+   */
   constructor(hotInstance) {
     super(hotInstance);
```

Comment:
Through this tutorial we use `this.hot` few times. But we never explain where it comes from. Maybe we should section or an explanation that BasePlugin provides few properties for us to use?

There are index mappers as well, we may link those to the index mapper description to keep this page short

Reply:
Done! However, I cannot link to the index-mapper part in Core (API ref) because the code snippet does not convert shortcodes `[...](...)` to links.